libs/crypto/include/crypto/stream_hasher.hpp

```diff
@@ -59,6 +61,11 @@ class StreamHasher
     return Update(reinterpret_cast<uint8_t const *>(vect.data()), vect.size() * sizeof(T));
   }
 
+  UInt256 Update(UInt256 const &val)
```

Comment:
This is not a great fit for this generic interface. While it is true that we only support SHA256 currently, this will change in the future.

Reply:
I am happy to make this a templated function of the number of bits in `UInt`, Note this is an overloaded `Update` function, not the `Final` function and it is neither `virtual`. Hence, it is not tied to SHA256 specifically and is just a convenience function.

Reply:
Updated in latest to any `UInt<S>`.